courses/management/commands/sync_courseruns.py

```diff
@@ -0,0 +1,86 @@
+"""
+Management command to sync grades and certificates for a course run
+"""
+from django.core.management.base import BaseCommand, CommandError
+
+from courses.models import CourseRun
+from courseware.api import get_edx_api_client
+from users.models import User
+from mitxpro.utils import now_in_utc
+
+
+class Command(BaseCommand):
+    """
+    Command to sync grades and certificates for a course run.
+    """
+
+    help = "Sync dates and title for all or a specific course run from edX."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--run",
+            type=str,
+            help="The 'courseware_id' value for a CourseRun",
+            required=False,
+        )
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):  # pylint: disable=too-many-locals
+        """Handle command execution"""
+        runs: [CourseRun] = []
+        if options["run"]:
+            try:
+                runs = [CourseRun.objects.get(courseware_id=options["run"])]
+            except CourseRun.DoesNotExist:
+                raise CommandError(
+                    "Could not find run with courseware_id={}".format(options["run"])
+                )
+        else:
+            now = now_in_utc()
+            runs = CourseRun.objects.filter(end_date__gt=now)
+
+        # TODO: This will be updated to use a service worker account
+        user = User.objects.filter(is_superuser=True).first()
+        api_client = get_edx_api_client(user).course_detail
+
+        success_logs = []
+        error_logs = []
+
+        # Iterate all eligible runs and sync if possible
+        for run in runs:
+            try:
+                course_detail = api_client.get_detail(run.courseware_id)
+
+                # Reset the expiration_date so it is calculated automatically and
+                # does not raise a validation error now that the start or end date
+                # has changed.
+                if (
+                    run.start_date != course_detail.start
+                    or run.end_date != course_detail.end
+                ):
+                    run.expiration_date = None
+
+                run.title = course_detail.name
+                run.start_date = course_detail.start
+                run.end_date = course_detail.end
+                run.enrollment_start = course_detail.enrollment_start
+                run.enrollment_end = course_detail.enrollment_end
+                try:
+                    run.save()
+                    success_logs.append(
+                        self.style.SUCCESS(
+                            "{}: Updated course run".format(run.courseware_id)
+                        )
+                    )
+                except Exception as e:  # pylint: disable=broad-except
```

Comment:
Is it possible to avoid broad-except?

Reply:
We want to capture all exceptions to Sentry, including network, connectivity, disk issues, cursor issues and whatnot. It won't be possible to exhaustively list all of them here, hence the broad except.